frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,471 @@
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
+import { InstallPlanPreview } from './install-plan';
+
+const getCRDFromObject = (obj: ClusterServiceVersionKind | InstallPlanKind | SubscriptionKind) => {
+  return obj?.metadata?.annotations?.['operatorframework.io/initialization-resource'];
+};
+
+const ViewInstalledOperatorsButton: React.FC<InstallPageMessageButtonProps> = ({
+  namespace,
+  className,
+}) => (
+  <Button
+    className={className}
+    data-test="view-installed-operators-btn"
+    variant="link"
+    component="a"
+    href={resourcePathFromModel(ClusterServiceVersionModel, null, namespace)}
+  >
+    View Installed Operators in{' '}
+    {namespace ? (
+      <>
+        namespace <b>{namespace}</b>
+      </>
+    ) : (
+      <b>all namespaces</b>
+    )}
+  </Button>
+);
+
+const InstallFailedMessage: React.FC<InstallPageMessageProps> = ({ namespace, csvName, obj }) => {
+  const crdMessage = getCRDFromObject(obj)
+    ? "The required custom resource can be created in the Operator's details view."
+    : '';
+
+  return (
+    <>
+      <h2 className="co-clusterserviceversion-install__heading">Operator installation failed</h2>
+      The operator did not install successfully. {crdMessage}
+      <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+        <Button
+          variant="primary"
+          component="a"
+          href={resourcePathFromModel(ClusterServiceVersionModel, csvName, namespace)}
+        >
+          View Error
+        </Button>{' '}
+        <ViewInstalledOperatorsButton namespace={namespace} />
+      </ActionGroup>
+    </>
+  );
+};
+
+const InstallNeedsApprovalMessage: React.FC<InstallPageMessageProps> = ({
+  namespace,
+  obj,
+  approve,
+}) => (
+  <>
+    <h2 className="co-clusterserviceversion-install__heading">Manual approval required</h2>
+    <p>
+      Review the manual install plan. Once approved, the following resources will be created in
+      order to satisfy the requirements for the components specified in the plan. Click the resource
+      name to view the resource in detail.
+    </p>
+    <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+      <Button variant="primary" onClick={approve}>
+        Approve
+      </Button>
+      <Button
+        className="co-clusterserviceversion__button"
+        variant="secondary"
+        component="a"
+        href={`${resourcePathFromModel(
+          SubscriptionModel,
+          obj?.metadata?.name,
+          namespace,
+        )}?showDelete=true`}
+      >
+        Deny
+      </Button>
+      <ViewInstalledOperatorsButton namespace={namespace} />
+    </ActionGroup>
+  </>
+);
+
+const CreateCRDButton: React.FC<InstallPageMessageButtonProps> = ({ className, obj, disabled }) => {
+  if (!getCRDFromObject(obj)) {
+    return null;
+  }
+  const crObj = safeLoad(getCRDFromObject(obj));
+  const crNamespace = crObj?.[0].metadata.namespace;
```

Comment:
If the `[0]?` is needed on the line below, it's needed here, too.

```suggestion
  const crNamespace = crObj?.[0]?.metadata.namespace;
```